docs/DeveloperGuide.adoc

```diff
@@ -337,17 +339,115 @@ Use case ends.
 * 3a. The given index is invalid.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1. DukeCooks shows an error message.
 +
 Use case resumes at step 2.
 
-_{More to be added}_
+[discrete]
+=== Use case: Suggest Food
+
+*MSS*
+
+1.  User requests for a food suggestion
+2.  DukeCooks gives a food recommendation
+3.  User is satisfied with the recommendation
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. User is unsatisfied with the food recommendations.
++
+[none]
+** 3a1. User asks for another suggestion.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Add Calorie Intake
+
+*MSS*
+
+1.  User requests to add calorie intake for the day
+2.  DukeCooks shows a list of Nutrition (Breakdown)
+3.  User requests to add certain nutrition value fields
+4.  DukeCooks updates Health Records
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The Nutrition list is empty (User did not give any nutrition inputs).
++
+[none]
+** 2a1. DukeCooks updates calorie count in Health Records (omit nutritional information).
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. DukeCooks shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Create Workout
+
+*MSS*
+
+1.  User requests to list <<exercise>>
+2.  DukeCooks shows a list of <<exercise>>
+3.  User requests to add an exercise into their <<workout>>
+4.  DukeCooks adds the exercise into the <<workout>>
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The exercise list is empty
++
+[none]
+** 2a1. DukeCooks updates the list with pre-made exercises.
++
+Use case resumes at step 2.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. DukeCooks shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Create a new diary
+
+*MSS*
+
+1.  User requests to create a new diary
+2.  A new diary is created with the specified name
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. Diary name already exists.
++
+[none]
+** 1a1. DukeCooks displays an error message.
++
+Use case ends.
 
 [appendix]
 == Non Functional Requirements
```

Comment:
Have you considered additional NFRs? For example, having backups in case of data corruption.